### DISCUSSION TITLE
CreateCompatibleDC(null) missing?

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [microsoft/CsWin32#604] [Q&A] [answered]
CreateComptibleDC needs an overload that takes null in order to get the applications window drawing context. There must be a way to get this but it's not obvious.

https://docs.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-createcompatibledc

**Comment by Gavin-Williams** [comment, marked as answer]
I think this works?
```
private HWND windowH;
...
HDC deviceContext = GetDC(windowH);
```

**Reply by Gavin-Williams** [reply]
So, for me, this works, I don't need CreateCompatibleDC

**Comment by AArnott** [comment]
I don't understand why you say the API is missing. When I ask for CreateCompatibleDC in NativeMethods.txt, it appears with methods that accept null or default(HDC).

**Reply by Gavin-Williams** [reply]
IntPtr dc = PInvoke.CreateCompatibleDC(null); // cannot convert from null to hdc

And I would never think to write default as the argument there. EVER. Just trying to communicate how such a signature is nearly undiscoverable.

**Reply by AArnott** [reply]
As of my last comment that you're replying to, what you wanted to do probably would have worked. But something evidently changed in the metadata or in CsWin32 that led to HDC parameters no longer leading to the creation of a `SafeHandle` friendly overload. This is tracked by #1049. When that's fixed, the syntax you tried will work again.
